vulcanai2/tests/models/test_dnn.py

```diff
@@ -86,3 +136,79 @@ def test_add_input_network(self, dnn_noclass, dnn_class):
         dnn_class.add_input_network(dnn_noclass)
         assert dnn_class.input_networks[dnn_noclass.name] is dnn_noclass
         assert dnn_class.in_dim == dnn_noclass.out_dim
+
+    # TODO: Failing fit! Fix the test
+    def test_fit_multi_input(self, multi_input_dnn,
+                             multi_input_train_loader,
+                             multi_input_test_loader):
+        """Test for fit function"""        
+        init_weights = copy.deepcopy(multi_input_dnn.network[0]._kernel.weight.detach())
+        multi_input_dnn_no_fit = copy.deepcopy(multi_input_dnn)
+        parameters1 = multi_input_dnn_no_fit.parameters()
+        try:
+            multi_input_dnn.fit(multi_input_train_loader, 
+                                multi_input_test_loader, 2)
+        except RuntimeError:
+            logger.error("The network multi_input_dnn failed to train.")
+        finally:
+            parameters2 = multi_input_dnn.parameters()
+            trained_weights = multi_input_dnn.network[0]._kernel.weight.detach().cpu()
+            
+            # Sanity check if the network parameters are training
+            assert (torch.equal(init_weights.cpu(), trained_weights.cpu()) is False)
```

Comment:
do you need to call `.cpu()` twice?

Reply:
that's a mistake. I will remove the repeated one.